lib/vagrant-cloudstack/action/run_instance.rb

```diff
@@ -205,7 +212,7 @@ def create_vm
                 :image_id => @template.id
             }
 
-            options['network_ids'] = @network.id unless @network.id.nil?
+            options['network_ids'] = @network_ids.join(",") unless @network_ids.empty?
```

Comment:
By not using `@network` you are effectively disabling specifying usage of `network_name`